libcudacxx/include/cuda/std/detail/libcxx/include/__config

```diff
@@ -1152,6 +1155,23 @@ typedef __char32_t char32_t;
 #endif
 #endif // _LIBCUDACXX_HAS_NO_LONG_DOUBLE
 
+#ifndef _LIBCUDACXX_HAS_NO_NVFP16
+#if !__has_include(<cuda_fp16.h>)                                               \
+  || (!defined(_LIBCUDACXX_COMPILER_CLANG_CUDA) && defined(_LIBCUDACXX_CUDACC_BELOW_12_2)) \
+  || (defined(_LIBCUDACXX_COMPILER_CLANG_CUDA) && CUDA_VERSION < 12020)         \
+  || (!defined(_LIBCUDACXX_CUDACC) && !defined(LIBCUDACXX_ENABLE_HOST_NVFP16))
+#  define _LIBCUDACXX_HAS_NO_NVFP16
+#endif
+#endif // _LIBCUDACXX_HAS_NO_NVFP16
+
+#ifndef _LIBCUDACXX_HAS_NO_NVBF16
+#if defined(CUB_DISABLE_BF16_SUPPORT)                                           \
+  || defined(_LIBCUDACXX_HAS_NO_NVFP16)                                         \
+  || !__has_include(<cuda_bf16.h>)
+#  define _LIBCUDACXX_HAS_NO_NVBF16
+#endif
+#endif // _LIBCUDACXX_HAS_NO_NVBF16
```

Comment:
I strongly dislike libcu++ precedent of double negation.

I would prefer if we could just name those `_LIBCUDACXX_HAS_NVFP16` and then check whether the macro is defined.

Furthermore we are always guarding this behind `__cuda_std__` which creates a lot of boilerplate. Can we just add that here and then only ever check against `_LIBCUDACXX_HAS_NVFP16` and `_LIBCUDACXX_HAS_NVBF16`

Reply:
Sure, don't really have strong feelings here.

Reply:
Oh, hang on. Those ifndef checks around the definitions **allow users to predefine the macros**, and, in effect, **disable** a feature forcibly. The positive macros don't give that ability; there's now no opt out.

I think the precedent (coming from upstream) is actually correct here.

Reply:
(Yes, they are using reserved identifiers, so it's technically illegal to tinker with them, but eh, that's a minor point in this conversation I think.)

Reply:
I believe this is backwards.

The internal Identifier should be easy to understand and avoid double negation. Any user interaction should come through non-ugly flags like ``